doc/CONFIGURATION.md

```diff
@@ -0,0 +1,289 @@
+
+The connection encryption protocols available are:
+
+- [libp2p/js-libp2p-secio](https://github.com/libp2p/js-libp2p-secio)
+
+If none of the available connection encryption mechanisms fulfills your needs, you can create a libp2p compatible one. A libp2p connection encryption protocol just needs to be compliant with the [interface-crypto](https://github.com/libp2p/js-interfaces/tree/master/src/crypto).
+
+If you want to get to know more about libp2p connection encryption, you should read the following content:
+
+- https://docs.libp2p.io/concepts/secure-comms
+- https://github.com/libp2p/specs/tree/master/connections
+
+### Peer Discovery
+
+> In a p2p network, peers must be able to discover other peers in the network.
+
+The peer discovery modules available are:
+
+- [js-libp2p-mdns](https://github.com/libp2p/js-libp2p-mdns)
+- [js-libp2p-bootstrap](https://github.com/libp2p/js-libp2p-bootstrap)
+- [js-libp2p-kad-dht](https://github.com/libp2p/js-libp2p-kad-dht)
+- [js-libp2p-webrtc-star](https://github.com/libp2p/js-libp2p-webrtc-star)
+- [js-libp2p-websocket-star](https://github.com/libp2p/js-libp2p-websocket-star) (Work in Progress)
+
+If none of the available peer discovery protocols fulfills your needs, you can create a libp2p compatible one. A libp2p peer discovery protocol just needs to be compliant with the [interface-peer-discovery](https://github.com/libp2p/js-interfaces/tree/master/src/peer-discovery).
+
+If you want to get to know more about libp2p peer discovery, you should read the following content:
+
+- https://github.com/libp2p/specs/blob/master/discovery/mdns.md
+
+### Content Routing
+
+> Content routing provides a way to find where content lives in the network. It works in two steps: 1) Peers provide (announce) to the network that they are holders of specific content and 2) Peers issue queries to find where that content lives. A Content Routing mechanism could be as complex as a DHT or a simple registry somewhere in the network.
+
+The content routing modules available are:
+
+- [js-libp2p-kad-dht](https://github.com/libp2p/js-libp2p-kad-dht)
+- [js-libp2p-delegated-peer-routing](https://github.com/libp2p/js-libp2p-delegated-peer-routing)
+
+If none of the available content routing protocols fulfills your needs, you can create a libp2p compatible one. A libp2p content routing protocol just needs to be compliant with the [interface-content-routing](https://github.com/libp2p/js-interfaces/tree/master/src/content-routing). **(WIP: This module is not yet implemented)**
+
+If you want to get to know more about libp2p content routing, you should read the following content:
+
+- https://docs.libp2p.io/concepts/content-routing
+
+### Peer Routing
+
+> Peer Routing offers a way to find other peers in the network by intentionally issuing queries, iterative or recursive, until a Peer is found or the closest Peers, given the Peer Routing algorithm strategy, are found.
```

Comment:
```suggestion
> Peer Routing offers a way to find other peers in the network by issuing queries using a Peer Routing algorithm, which may be iterative or recursive. If the algorithm is unable to find the target peer, it will return the peers that are "closest" to the target peer, using a distance metric defined by the algorithm.
```